### ISSUE TITLE
There is no way to update jruby-jars to the latest snapshot build

### ISSUE THREAD

**Issue by cshupp1** [jruby/jruby#5822]
There is no easy way to test your defects in the latest snapshot build as there is no mechanism
to do a 'bundle update jruby-jars@snapshot'

This ability would make verification of defects easier for all of us.

Thanks,

Cris

**Comment by headius** [comment]
@mkristian @enebo Ideas?

Obviously we could start building the gem and pushing it as a Maven artifact. That would at least make the file accessible.

However if the dependency is through a Ruby mechanism, like a Gemfile, I don't think this will work; we can't point at Maven central and have RubyGems resolve our snapshot releases (right?).

Perhaps we could make a git repository that looks like the jruby-jars gem but always fetches the snapshot jars from sonatype on install?

Perhaps we could push a jruby-jars-9.2.8.0.snapshot gem for each in-progress release that also fetches latest snapshot jars via jar-dependencies?

I'm not sure what's the best direction here.

**Comment by enebo** [comment]
nightly pushing to rubygems itself via ci?  We are a pretty large gem so I am not sure how much that would be liked (although we are vagrant :) ).  I don't actually know how others do this for gems?  I do not want to maintain an image for our own gemserver but I would imagine most people would also be in that boat.

**Comment by kares** [comment]
believe we could setup a hook to do the following on master commits:
- checkout the repo and try building the jruby-jars maven artefact
- if that goes well -> rename the jar as needed -> *9.X.Y.Z.SNAPSHOT*
- update the .jar file in  a (known) :git repo (such as `jruby/jruby-jars`)
- commit the other repo

the repo would need to have a gemspec and few files (of the expanded gem archive). there's a risk of it not working when jruby-jars.gem changes anything (due duplication). another downside is commits won't be matching actual jruby changes but the commit could include the originating commit sha. 

also the build could probably be going on the jruby-jars repo, thus on jruby's end it is just triggering a webhook and the build does not get complicated.

believe this could be a nice task for someone caring of having jruby-jars snapshots, if its agreed that there isn't a better way.

**Comment by headius** [comment]
@kares Yeah that's what I was thinking. An alternative would be to have the jruby-jars gem use jar-dependencies to fetch the libraries; then while a given JRuby release is still in development, we'd just push a snapshot gem that knows to go to the snapshot repo. The only updates needed for the jruby-jars repo then would be to point it at a release, tag that as the gem release, and then point it at the next snapshot for ongoing dev.

**Comment by kares** [comment]
updating and tagging the second repo sounds like more release work ... but likely a bot could do that as well. esp. since the real release happens from this repo.

**Comment by mkristian** [comment]
actually we skip to deploy the jruby-jars-x.x.x.x.SNAPSHOT.gem to be deployed to maven-central. this should be easy to add. not if this helps as some simple form of rubygems repo is needed to use with `gem` or `bundle`

regarding jar-dependencies: the question is if we end up with the same directory structure as current gem, not sure if people really use the `JRubyJars.core_jar_path` and `JRubyJars.stdlib_jar_path` provided by the gem or if we can get this two paths work correctly with jars-dependencies.

**Comment by mohits-qi** [comment]
It's not ideal but I follow this method:
* rebuild jruby-jars locally
* install the jruby-jars gem
...use!

It does require you to rebuild the snapshot locally (at least for now, from what I can tell) but this works for me. I've added this to a post here: https://notepad.onghu.com/2021/jruby-use-new-jruby-jars/

**Comment by mohits** [comment]
It's not ideal but I follow this method:

    rebuild jruby-jars locally
    install the jruby-jars gem
    ...use!

It does require you to rebuild the snapshot locally (at least for now, from what I can tell) but this works for me. I've added this to a post here: https://notepad.onghu.com/2021/jruby-use-new-jruby-jars/
